Graphics/Slicer/Math/PGA.hs

```diff
@@ -235,12 +299,14 @@ ppointToPoint2 (PPoint2 (GVec vals)) = if infinitePoint
     yVal =          valOf 0 $ getVals [GEZero 1, GEPlus 1] vals
     infinitePoint = 0 == valOf 0 (getVals [GEPlus 1, GEPlus 2] vals)
 
--- | Create a 2D projective line from a pair of euclidian endpoints.
+-- | Create a projective line from a euclidian line segment.
 eToPLine2 :: LineSeg -> PLine2
-eToPLine2 (LineSeg (Point2 (x1,y1)) (Point2 (x,y))) = PLine2 $ GVec $ foldl addVal [] [ GVal c [GEZero 1], GVal a [GEPlus 1], GVal b [GEPlus 2] ]
+eToPLine2 (LineSeg startPoint@(Point2 (x1,y1)) (Point2 (x,y))) = plineFromEndpoints startPoint (Point2 (x1+x,y1+y))
+
+-- | Create a projective line from a pair of euclidian points.
+plineFromEndpoints :: Point2 -> Point2 -> PLine2
+plineFromEndpoints (Point2 (x1,y1)) (Point2 (x2,y2)) = PLine2 $ GVec $ foldl addVal [] [ GVal c [GEZero 1], GVal a [GEPlus 1], GVal b [GEPlus 2] ]
```

Comment:
**Always** use `foldl'` instead of `foldl`. The latter will always leak memory for absolutely no gain. It's a bug in the standard library.

Reply:
yeah, i knew this, just forgot it.